src/wasm/key_utils.rs

```diff
@@ -191,39 +166,37 @@ pub(crate) fn sign_with_key(k: &RistrettoSecretKey, e: &[u8], r: Option<&Ristret
 }
 
 /// Checks the validity of a Schnorr signature
-#[allow(non_snake_case)]
 #[wasm_bindgen]
 pub fn check_signature(pub_nonce: &str, signature: &str, pub_key: &str, msg: &str) -> JsValue {
-    let mut result = SignatureVerifyResult {
-        result: false,
-        error: "".into(),
-    };
+    let mut result = SignatureVerifyResult::default();
 
-    let R = match RistrettoPublicKey::from_hex(pub_nonce) {
+    let pub_nonce = match RistrettoPublicKey::from_hex(pub_nonce) {
         Ok(n) => n,
         Err(_) => {
             result.error = format!("{} is not a valid public nonce", pub_nonce);
             return JsValue::from_serde(&result).unwrap();
         },
     };
 
-    let P = RistrettoPublicKey::from_hex(pub_key);
-    if P.is_err() {
-        result.error = format!("{} is not a valid public key", pub_key);
-        return JsValue::from_serde(&result).unwrap();
-    }
-    let P = P.unwrap();
+    let pub_key = match RistrettoPublicKey::from_hex(pub_key) {
+        Ok(p) => p,
+        Err(_) => {
+            result.error = format!("{} is not a valid public key", pub_key);
+            return JsValue::from_serde(&result).unwrap();
+        },
+    };
 
-    let s = RistrettoSecretKey::from_hex(signature);
-    if s.is_err() {
-        result.error = format!("{} is not a valid hex representation of a signature", signature);
-        return JsValue::from_serde(&result).unwrap();
-    }
-    let s = s.unwrap();
+    let s = match RistrettoSecretKey::from_hex(signature) {
+        Ok(s) => s,
+        Err(_) => {
+            result.error = format!("{} is not a valid hex representation of a signature", signature);
+            return JsValue::from_serde(&result).unwrap();
+        },
+    };
 
-    let sig = RistrettoSchnorr::new(R, s);
+    let sig = RistrettoSchnorr::new(pub_nonce, s);
```

Comment:
P and R all well-known conventions for Pubkey and nonce. Please don't unilaterally make the decision to rename variables.

Reply:
Happy to change it if given the chance as part of the PR process (i.e. not unilateral 😄 ☮️ ), really just getting rid of `#[allow(non_snake_case)]`